src/bash/sdv-provision

```diff
@@ -112,6 +118,116 @@ showDeviceCerts() {
 
 loadDeviceId
 
+
+# WHen the configuration has changed, we need to recreate the container
+recreateCloudConnectorContainer() {
+    echo "Recreating the Cloud Connector container..."
+    kanto-cm stop -n cloudconnector -f --timeout 5
+    kanto-cm remove -n cloudconnector
+    systemctl restart kanto-auto-deployer
+}
+
+
+# For the cloud connection to know which backend to use, we either need
+# - id_scope -> connect to global Azure IoT Device Provisioning Service and use id_scope to know which tenant
+# - ConnectionString -> connect to Hostname using DeviceID. When using certificates, the connectionstring is not supposed to contain a SharedAccessKey
+
+configureIdScope() {
+    while true; do
+        read -p "Please enter your Id Scope of the Azure IoT Device Provisioning Service:" idscope
+        if [ -n ${idscope} ]; then
+            echo ${idscope} > ${DATA_AZURE_ID_SCOPE_FILE}
+
+            # Modify the cloudconnector.json deployment descriptor
+            local sourcejson="${CLOUDCONNECTOR_DEPLOYMENT_DESCRIPTOR_TEMPLATE}"
+            if [ ! -f ${CLOUDCONNECTOR_DEPLOYMENT_DESCRIPTOR_TEMPLATE} ]; then
+                sourcejson=${CLOUDCONNECTOR_DEPLOYMENT_DESCRIPTOR}
+            fi
+            local targetjson="${CLOUDCONNECTOR_DEPLOYMENT_DESCRIPTOR}"
+            local tmp=$(mktemp -t leda-idscope-XXXXXX)
+            
+            jq ".config.env += [\"ID_SCOPE=$idscope\"]" "${sourcejson}" > "$tmp" && mv "$tmp" "${targetjson}"
```

Comment:
This line is hard to understand if ID_SCOPE= does not exist

Reply:
I'll add documentation to it.

The ID_SCOPE environment variable entry is added to the .json file by using jq. It will keep all the existing environment variables though. Downside of this is, that if the script is rerun in-place with the same .json file (e.g. not using the .json.template), then the environment variable may be added twice (duplicate).